### ISSUE TITLE
[Bug][a11y]: Card with href link ordering creates redundancy and disabled important link functionality

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3385]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Card

### Expected behavior

Within a Card with an href, the link should come before the Card body, and the title for the Card should be hidden from assistive technology using `aria-hidden="true"`. 

A user should be able to right click or activate a context menu to open the link in a new Tab.

It shouldn't require two clicks on the Card to open the link.

### Actual behavior

A Card with an href adds a link that follows the card body and repeats the title, which adds unnecessary verbosity.  

The link css declares `pointer-events: none;`, which disables the users' ability to open the link in a new tab, and makes it so that sometimes the user has to click the Card twice in order for it to open the link.

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error

**Comment by caseyisonit** [comment]
Hello @jnurthen in order to help us triage your issue, [can you please review our bug severity classification guide](https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md#issue-severity-classification) and provide us with the appropriate severity classification? Please update the label to the correct severity level and remove `SEV missing` label.

If we don't hear from you in 2 weeks, we will close this issue as not planned.

<img width="1024" height="1028" alt="Image" src="https://github.com/user-attachments/assets/d5fd78ee-6e8e-4967-97f3-4e4198b79f13" />
